src/Type/ClosureType.php

```diff
@@ -75,6 +76,7 @@ public function __construct(
 		?TemplateTypeMap $templateTypeMap = null,
 		?TemplateTypeMap $resolvedTemplateTypeMap = null,
 		?TemplateTypeVarianceMap $callSiteVarianceMap = null,
```

Comment:
Please check the commit where it was added, and what it does in ClosureType, that should give you some context - and do the same in CallableType :)

Reply:
reading https://phpstan.org/blog/guide-to-call-site-generic-variance i'm not sure call-site variance would apply to ClosureType / CallableType unless ClosureType is created as a firstClassCallable, so it's correct for the variance map to be empty, unless @jiripudil has wisdom for us? :angel:

Reply:
Hi, that's right, ClosureType is only created with the call-site variance map when constructed from callable, and even then it's more for the sake of completeness than anything else.

In hindsight, it might actually be an oversight that it remained there: since I had to move the getCallSiteVarianceMap() method to a different interface for BC reasons (https://github.com/phpstan/phpstan-src/pull/2485/commits/107716730a1c1b1dd76fdc9b0aaa2950dd593d40), it will only be passed around the ClosureTypes but will never be called to do anything useful :|

Reply:
should i revert the change to add the variance map to CallableType then ?

Reply:
Yes, looks like it :)